### PR TITLE
Add new signup view

### LINKED ISSUE #1611: Sign up flow changes: Active consent to our Terms of Service
Regarding GDPR (General Data Protection Regulation) changes we will need 2 additions to current sign up flow: User’s Country of residence & active consent to our Terms of Service.

## Active consent to our Terms of Service
- “By signing up you agree to terms of service…” can no longer be just a passive text or pre-ticked checkbox. This means the user must “actively” consent, so e.g. tick a box or push some “Agree” button, in order to create an account. We decided to use two-step sign up and ask user to interact with "I agree" button.
- Confirmation with timestamp that they have done this must be sent to backend, they’ll modify the v9/signups endpoint accordingly https://github.com/toggl/backend/issues/6354
- Open this popup after user taps "Sign up for free" button in the Sign up view #1960
- Send the form to the BE after user has tapped "I agree" button
- User can close the popup by tapping the 'x' button or on the background. Then user will be directed back to the Sign up page. Fields that user filled in before must remain filled.
- ToS: https://toggl.com/legal/terms/
- PP: https://toggl.com/legal/privacy/

### ToS & PP popup
toggl - ios - sign up tos pp 1x

Zeplin: https://zpl.io/blmM4jk

### LINKED ISSUE #1960: Sign up flow changes: New Sign up UI
- New Sign up UI is accessible from the Log in page.
- New Sign up UI contains all needed fields and inputs: Email and Password field, Country selector, Google Sign up, and option to switch back to Log in page.
- By default, all the fields are not focused.
- Error messages are shown below the Password field (if the message is longer then in two rows).
- When the form can't be submitted, disable the main "Sign up for free" button. Text opacity = 48%
- Tapping Country dropdown at the top of the screen opens Country of Residence view #1610
- Form won't be submitted after tapping "Sign up for free", instead ToS & PP popup will be opened #1611 after which we can post it to BE.
- When Email or Password field is focused or when user has already typed something in, then the placeholder moves up.
- Bottom card is always positioned at the bottom of the screen and directs user back to the Log in page. The whole card should be tappable.
- Everything else is positioned at the top of the screen:
   - On iphone SE sized screens content starts 24pt from the nav bar (country dropdown)
   - On bigger screens content starts 72pt from the nav bar

### Sign up page
toggl - ios - sign up 1x

Zeplin: https://zpl.io/2vl6LKr

### Active field & Error message
toggl - ios - sign up keyboard error message 1x

Zeplin: https://zpl.io/VqxY8Wl

### PR DESCRIPTION
**Pull request by arturs105**
Closes #1960 
Closes #1611

Squash whenever.

### CODE REVIEW

File: Toggl.Daneel/Extensions/LoginSignupViewExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+            layer.ShadowPath = shadowPath.CGPath;
+            layer.ShadowRadius = 1;
+
+            //Add spacing between button title ang google logo
+            var spacing = 17;
+            button.ImageEdgeInsets = new UIEdgeInsets(0, 0, 0, spacing);
+            button.TitleEdgeInsets = new UIEdgeInsets(0, spacing, 0, 0);
+        }
+
+        public static void SetupBottonCard(this UIView card)
```
[the remark this answers is not in the document]
Comment (arturs105) [reply Toggl.Daneel/Extensions/LoginSignupViewExtensions.cs]: `Bottom` 😄

File: Toggl.Daneel/ViewControllers/TermsOfServiceViewController.cs [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+        }
+
+        private void onTextViewTapped(UITapGestureRecognizer recognizer)
+        {
+            var layoutManager = TextView.LayoutManager;
+            var location = recognizer.LocationInView(TextView);
+            location.X -= TextView.TextContainerInset.Left;
+            location.Y -= TextView.TextContainerInset.Top;
+
+            nfloat _ = 0;
```
[the remark this answers is not in the document]
Comment (arturs105) [reply Toggl.Daneel/ViewControllers/TermsOfServiceViewController.cs]: Yeah, i couldn't find another way.

File: Toggl.Foundation.MvvmCross/ViewModels/NewLoginViewModel.cs [hunk 3]
```diff
@@ -189,5 +199,8 @@ private void googleLogin()
                 .Do(_ => analyticsService.TrackLoginEvent(AuthenticationMethod.Google))
                 .Subscribe(onDataSource, onError, onCompleted);
         }
+
+        private Task signup()
+            => navigationService.Navigate<SignupViewModel>();
```
[the remark this answers is not in the document]
Comment (arturs105) [reply Toggl.Foundation.MvvmCross/ViewModels/NewLoginViewModel.cs]: We could, but it wasn't in the spec, so i went the lazy way.

[the remark this answers is not in the document]
Comment (simonrozsival) [reply Toggl.Foundation.MvvmCross/ViewModels/NewLoginViewModel.cs]: What do you think, @pe0ny?

[the remark this answers is not in the document]
Comment (pe0ny) [reply Toggl.Foundation.MvvmCross/ViewModels/NewLoginViewModel.cs]: Yes, we should.

**Pull request merged**